### ISSUE TITLE
question related to moordyn

### ISSUE THREAD

**Issue by WANGSUYU0809** [OpenFAST/openfast#2205]
Hi,

I have some questions related to Moordyn for modeling mooring systems.

Firstly, how to specify mudline postion  (where the mooring lines lay on top of) ? Does it equal to 'WtrDpth' defined in  lines.txt?

Secondly, what does the parameter ' kBot' defined in  lines.txt represent? Does it represent vertical contact stiffness between mooring line and mudline?  If I set this parameter to 0, does it mean there would be no mudline to support mooring lines?

Best Regards,
Wang Suyu

**Comment by WANGSUYU0809** [comment]
Besides, another related question is, how to include friction between mudline and mooring lines in Moordyn?

**Comment by RyanDavies19** [comment]
@WANGSUYU0809 Check out the MoorDyn documentation input file [options section](https://moordyn.readthedocs.io/en/latest/inputs.html#id4). I also would suggest reading this theory paper, which describes the seabed contact models used in MoorDyn: https://www.nrel.gov/docs/fy23osti/82033.pdf

**Comment by WANGSUYU0809** [comment]
Thanks for your answer! From the documents you shared, it seems that the mudline friction is only considered in Moordyn V2, but not considered in Moordyn V1.  Am I right?

Besides, I have a further question related to definitions in 'lines.txt':

When we define the MassDen in LINE TYPES, is this dry mass? (mass in air, buoyancy not considered)  And I guess the buoyancy of mooring lines will be calculated additionaly by the diameter 'D' defined in LINE TYPES, basded on formula pi()/4*D^2*1024. So that the 'D' should be defined coresponding to a circular cross-section, but could represent equivalent displacement with the real mooring chains, am I right?
Best Regards,

**Comment by RyanDavies19** [comment]
@WANGSUYU0809 yes friction was added in MoorDyn v2. In general users should only use MoorDyn v2, as it has been improved in terms of accuracy and reliability from MoorDyn v1. 

Secondly, yes it is the dry mass. All of the mass inputs to MoorDyn are mass in air, and MoorDyn uses the objects geometry to find buoyancy. In the case of lines, the volume of each segment is calculated and used with the mass of each segment to calculate the weight in water. For a cylindrical cable/line this would be the cross sectional diameter, for a chain it would be the volume equivalent diameter.

**Comment by WANGSUYU0809** [comment]
> @WANGSUYU0809 yes friction was added in MoorDyn v2. In general users should only use MoorDyn v2, as it has been improved in terms of accuracy and reliability from MoorDyn v1.
> 
> Secondly, yes it is the dry mass. All of the mass inputs to MoorDyn are mass in air, and MoorDyn uses the objects geometry to find buoyancy. In the case of lines, the volume of each segment is calculated and used with the mass of each segment to calculate the weight in water. For a cylindrical cable/line this would be the cross sectional diameter, for a chain it would be the volume equivalent diameter.

Thanks for your answer, just for final confirmation:

Based on my understading, we should define dry mass in lines.txt, for the Diameter D in lines.txt:
1. For cylindrical cable/line, we could input the actual diameter of the cable/line directly,
2. For chain, we should input 'volume equivalent diameter'. So that for the hydrodynamic Cd and Ca coefficients, we should also make adjustment, because they are also  based on ‘volume equivalent diameter’.

Am I  correct?

Thanks a lot.

**Comment by RyanDavies19** [comment]
@WANGSUYU0809 yes you are correct

**Comment by WANGSUYU0809** [comment]
Thanks for your help, it clears my question!
